Purchases/PurchasesExtensions/PurchaseOwnershipType+Extensions.swift

```diff
@@ -0,0 +1,33 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PurchaseOwnershipType+Extensions.swift
+//
+//  Created by Juanpe Catalán on 26/8/21.
+
+import Foundation
+
+extension PurchaseOwnershipType: Decodable {
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.singleValueContainer()
+        let purchaseOwnershipTypeString = try container.decode(String.self)
+
+        switch purchaseOwnershipTypeString {
+        case "PURCHASED":
+            self = .purchased
+        case "FAMILY_SHARED":
+            self = .familyShared
+        default:
+            Logger.warn("received unknown ownershipType: \(purchaseOwnershipTypeString)")
+            self = .unknown
```

Comment:
I was working on this, and I realized that there is a special case, and I don't know exactly why. So I'd rather ask before assuming something wrongly 🙂

For both the `periodType` and the `Store` type, the default value is the same when the key is missing and when the value is unknown. But, this behaviour is different for `PurchaseOwnershipType`. If the key is missing, the default value is `purchased`, but if the value is unknown, the default value is `unknown`.

The question is, we could use a helper method to decode the value for the key and define the default value as well for both the `periodType` and the `Store` type and use the code that belongs to this PR for `PurchaseOwnershipType` or use the helper method for all the keys and simplify the logic for `PurchaseOwnershipType`.

WDYT? 🤔

Reply:
We recently were talking about this here:
https://github.com/RevenueCat/purchases-ios/issues/775

The TL;DR: is that `ownershipType` is new, and before it existed, you could only just own something by purchasing it directly, so if the value is empty we assume that it was a direct purchase. 
But if we get a value that's different from the ones we know about, we might have to assume that there's a new way to own something (like if apple introduced a bulk purchase program for businesses, for example). 
We might wanna have an `@unknown default` for that case in case something new comes up and we forgot to add it here. 
As for the best way to handle both cases here, I don't have a strong preference, so I'm happy to go with whatever feels best to you.

Reply:
OK, understood it. I gonna submit the changes that includes my proposal that I think could fit well the current scenario 👍🏼